plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -271,8 +253,301 @@ type Query {
 }
 `;
 
-// Pass implementation of MetronomeClient to the resolvers for easier testability
-export const resolvers = (): IResolvers => ({
+interface ResolverArgs {
+  fetchJobDetail: (id: string) => Observable<MetronomeClient.JobDetailResponse>;
+  pollingInterval: number;
+}
+
+function isActiveRun(arg: any): arg is MetronomeClient.ActiveJobRun {
+  return arg.jobId !== undefined;
+}
+
+function isHistoryRun(arg: any): arg is MetronomeClient.JobHistoryRun {
+  return arg.id !== undefined;
+}
+
+const typeResolvers = {
+  Job(response: MetronomeClient.JobDetailResponse): Job {
+    return {
+      id: response.id,
+      description: response.description,
+      command: response.run.cmd,
+      disk: response.run.disk,
+      mem: response.run.mem,
+      cpus: response.run.cpus,
+      docker: fieldResolvers.Job.docker(response),
+      json: fieldResolvers.Job.json(response),
+      labels: fieldResolvers.Job.labels(response),
+      lastRunStatus: fieldResolvers.Job.lastRunStatus(response),
+      name: fieldResolvers.Job.name(response),
+      jobRuns: fieldResolvers.Job.jobRuns(response),
+      lastRunsSummary: fieldResolvers.Job.lastRunsSummary(response),
+      scheduleStatus: fieldResolvers.Job.scheduleStatus(response),
+      activeRuns: typeResolvers.JobRunConnection(response.activeRuns),
+      schedules: fieldResolvers.Job.schedules(response)
+    };
+  },
+  JobRun(run: MetronomeClient.ActiveJobRun | JobHistoryRunWithStatus): JobRun {
+    return {
+      jobID: fieldResolvers.JobRun.jobID(run),
+      dateCreated: DateUtil.strToMs(run.createdAt),
+      dateFinished: fieldResolvers.JobRun.dateFinished(run),
+      status: run.status, // TODO: derive from where it comes from before passing to this function
```

Comment:
What is this todo abou?

Reply:
not that I'm aware of - maybe copied from the old implementation?

Reply:
might be 🤷‍♂️

Reply:
@Fabs is going to fix it on his branch